### DISCUSSION TITLE
deploy jsf web app to weblogic cause "<BEA-160237> <Failed to load class"

### DISCUSSION THREAD

**Discussion by GerardoRS** [primefaces/community#1154] [PrimeFaces]
Hi, I have a jsf 2.2 and primefaces 6.2 aplication, when i deploy it to weblogic, the log shows lot of "<BEA-160237> <Failed to load class" errors reffering to libraries that are not needed in the application, in other words the application works, but i get lot of deployment errors, I read about "metadata-complete="true" but it doesn't works, and the weblogic server is not restarting ok after deploying that application, any suggestions?

**Comment by melloware** [comment]
You didn't list any examples of the classes it says it can't find?  I somehow doubt this is PrimeFaces related...

**Reply by GerardoRS** [reply]
Yes, sorry, this is the first message in the log, there are many others:

<BEA-160237> <Failed to load class org.atmosphere.annotation.Processor by ClassLoader weblogic.utils.c7 finder: weblogic.utils.classloaders.CodeGenClassFinder@4cdcb04a annotation:  when looking up impelementation classes for classes specified by the value of @HandleTypes. The class loading error.Processor. Class org.atmosphere.annotation.Processor will be ignored for processing the @HandleTypes annotation.>

I just created another project using just jsf 2.2 and i'm not getting the errors in the log while deploying!

**Reply by melloware** [reply]
Atmosphere is the old Push library which has been removed from PF a long time ago.  I see you are using 6.2 which is quite old.

**Reply by GerardoRS** [reply]
yes there are messages for groovy, kotlin, aspectj, jaxen, springframework.context, etc, the thing is that in netbeans the project compiles ok, and the application works without problems in the server, I'm not using any of those classes the log indicates, and the admin team is having problems when they restart any of the weblogic servers because they get stuck and when they undeploy this application the server restarts ok!
